### PR TITLE
Allow fsgroup/SELinux attributes to be set

### PR DESCRIPTION
**Pull request by azdagron**
- Changes the mount to be mounted read-write on the host so that
  fsetxattr can be used by the host to change the attributes on files
  inside the mount. For security purposes, this only happens if the CSI
  volume is specified as read-only so the kubelet will mount the volume
  read-only into the containers.
- Optionally enforces that the CSI volume is marked read-only. We can't
  enforce this by default, since it would break existing deployments.
  It will be enforced in a future release.

Fixes: #42

### CODE REVIEW

File: cmd/spiffe-csi-driver/main.go [hunk 1]
```diff
@@ -39,19 +40,32 @@ func main() {
 	}
 	log = zapr.NewLogger(zapLog)
 
+	// If the enforce-read-only-volume flag was not explicitly provided,
+	// instruct the user to set it. Either way, if it is set to false, or
+	// unset, emit a log educating users that the will be enforced in the
+	// future.
+	// TODO: enforce this by default in a future release.
+	if !isFlagSet("enforce-read-only-volume") {
+		log.Error(nil, "Pass the --enforce-read-only-volume flag to enforce that the CSI volume is marked read-only")
```
Comment (erikgb) [inline cmd/spiffe-csi-driver/main.go:49]: Nit: Should this be logged at the warn level instead? Or is the error level intentional?

File: cmd/spiffe-csi-driver/main.go [hunk 2]
```diff
@@ -39,19 +40,32 @@ func main() {
...
+	}
+	if !*enforceReadOnlyVolumeFlag {
+		log.Error(nil, "Not enforcing that the CSI volume is marked read-only; this will be required in a future release")
```
Comment (erikgb) [inline cmd/spiffe-csi-driver/main.go:52]: Nit: Should this be logged at the warn level instead? Or is the error level intentional?

**Review by erikgb — APPROVED** [review]
LGTM, but a couple of related nits/questions. Good job!

File: cmd/spiffe-csi-driver/main.go (hunk 1 above)
Comment (azdagron) [reply cmd/spiffe-csi-driver/main.go:49]: For better or worse, logr has become somewhat of the defacto logger interface in k8s projects and `logr` doesn't believe in log levels. It's either informational (Info) or there is a problem (Error). There is a [philosophy](https://github.com/go-logr/logr#why-not-named-levels-like-infowarningerror) behind it.

Comment (erikgb) [reply cmd/spiffe-csi-driver/main.go:49]: True, I forgot about that. 😉 Please ignore/resolve my two comments about this.

Comment (azdagron) [reply cmd/spiffe-csi-driver/main.go:49]: I forgot too. The first thing I typed was `log.Warn` :laughing: and my compiler had to remind me.

**Review by MarcosDY — APPROVED** [review]
LGTM!!!

**Pull request merged**